src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala

```diff
@@ -22,10 +22,29 @@ object RouteName {
 
 case class TeamName(name: String) extends AnyRef
 
+case class UserName(name: String) extends AnyRef {
+
+  def apply(name: Option[String]) = name match {
+    case Some(name) => UserName(name)
+    case _          => throw InvalidUsereNameException
+  }
+}
+
+object UserName {
+  def apply(name: Option[String]): UserName = name match {
```

Comment:
This is only an overriden constructor... So you are able to pass both a string or an option.

I removed the `apply` method in the case class, as it's useless.